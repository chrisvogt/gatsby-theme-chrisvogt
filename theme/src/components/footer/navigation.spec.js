import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from './navigation'
import useNavigationData from '../../hooks/use-navigation-data'

jest.mock('../../hooks/use-navigation-data')

describe('Footer/Navigation', () => {
  useNavigationData.mockImplementation(() => ({
    footer: [
      {
        slug: 'about-me',
        path: '/about',
        title: 'About Me',
        text: 'About Me'
      }
    ]
  }))

  it('matches the snapshot', () => {
    const tree = renderer.create(<Navigation />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders hyperlinks for the navigation items', () => {
    const testRenderer = renderer.create(<Navigation />)
    const testInstance = testRenderer.root
    expect(testInstance.findAllByType('a').length).toBe(1)
  })
})
