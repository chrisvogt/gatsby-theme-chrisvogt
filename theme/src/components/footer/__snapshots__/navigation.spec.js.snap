// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer/Navigation matches the snapshot 1`] = `
<a
  className="css-1ay9vb9"
  href="/about"
  title="About Me"
>
  About Me
</a>
`;
